python/pyarrow/table.pxi

```diff
@@ -1514,6 +1514,30 @@ cdef class Table(_PandasConvertible):
 
         return pyarrow_wrap_table(c_table)
 
+    @property
+    def column_names(self):
+        """
+        Names of the table's columns
+        """
+        names = self.table.ColumnNames()
+        return [name for name in names]
+
+    def rename_columns(self, names):
+        """
+        Rename columns with provided names
```

Comment:
I think it's worth mentioning it returns a new Array like the previous methods, and add a very small test in `test_table`. Sorry if I missed this initially!

Reply:
no worries, I'll update now

Reply:
@fsaintjacques done, and thanks: this updated column names [but not the schema](https://github.com/apache/arrow/pull/4557/commits/213c19383b8dc7536fb1ce4c68a7fb70b5835d94#diff-74388640577855d01824e59501ea2791R501)